tool_registry/installers.py

```
@@ -105,17 +188,40 @@ def install_native_tools(
+                    _extract_compressed_binary(archive_path, source.archive_inner_path, binary_path)
+                finally:
+                    archive_path.unlink(missing_ok=True)
             else:
-                logger.warning("  %s: download failed (empty file)", dep.binary_name)
-                binary_path.unlink(missing_ok=True)
+                if not download_asset(url, binary_path, expected_sha256=expected_hash):
+                    logger.warning("  %s: download failed (empty file)", dep.binary_name)
+                    binary_path.unlink(missing_ok=True)
+                    continue
+            if system != "Windows":
+                os.chmod(binary_path, 0o755)
```

Ruff (0.15.9)

[error] `os.chmod` setting a permissive mask `0o755` on file or directory

(S103)